webapp/channels/src/components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx

```diff
@@ -21,74 +22,56 @@ export interface Props extends PropsFromRedux {
+        actions.leaveChannel(channel.id);
         trackEvent('ui', 'ui_public_channel_x_button_clicked');
         callback();
-    };
+    }, [channel.id]);
 
-    handleLeavePrivateChannel = (callback: () => void) => {
-        this.props.actions.openModal({modalId: ModalIdentifiers.LEAVE_PRIVATE_CHANNEL_MODAL, dialogType: LeaveChannelModal, dialogProps: {channel: this.props.channel}});
+    const handleLeavePrivateChannel = useCallback((callback: () => void) => {
+        actions.openModal({modalId: ModalIdentifiers.LEAVE_PRIVATE_CHANNEL_MODAL, dialogType: LeaveChannelModal, dialogProps: {channel}});
         trackEvent('ui', 'ui_private_channel_x_button_clicked');
         callback();
-    };
-
-    getChannelLeaveHandler = () => {
-        const {channel} = this.props;
-
-        if (channel.type === Constants.OPEN_CHANNEL && channel.name !== Constants.DEFAULT_CHANNEL) {
-            return this.handleLeavePublicChannel;
-        } else if (channel.type === Constants.PRIVATE_CHANNEL) {
-            return this.handleLeavePrivateChannel;
-        }
-
-        return null;
-    };
-
-    getIcon = () => {
-        const {channel} = this.props;
-
-        if (channel.shared) {
-            return (
-                <SharedChannelIndicator
-                    className='icon'
-                    channelType={channel.type}
-                    withTooltip={true}
-                />
-            );
-        } else if (channel.type === Constants.OPEN_CHANNEL) {
-            return (
-                <i className='icon icon-globe'/>
-            );
-        } else if (channel.type === Constants.PRIVATE_CHANNEL) {
-            return (
-                <i className='icon icon-lock-outline'/>
-            );
-        }
-
-        return null;
-    };
-
-    render() {
-        const {channel, currentTeamName} = this.props;
-
-        let ariaLabelPrefix;
-        if (channel.type === Constants.OPEN_CHANNEL) {
-            ariaLabelPrefix = localizeMessage('accessibility.sidebar.types.public', 'public channel');
-        } else if (channel.type === Constants.PRIVATE_CHANNEL) {
-            ariaLabelPrefix = localizeMessage('accessibility.sidebar.types.private', 'private channel');
-        }
-
-        return (
-            <SidebarChannelLink
-                channel={channel}
-                link={`/${currentTeamName}/channels/${channel.name}`}
-                label={channel.display_name}
-                ariaLabelPrefix={ariaLabelPrefix}
-                channelLeaveHandler={this.getChannelLeaveHandler()!}
-                icon={this.getIcon()!}
-            />
-        );
+    }, [channel]);
+
+    let channelLeaveHandler = null;
+    if (channel.type === Constants.OPEN_CHANNEL && channel.name !== Constants.DEFAULT_CHANNEL) {
+        channelLeaveHandler = handleLeavePublicChannel;
+    } else if (channel.type === Constants.PRIVATE_CHANNEL) {
+        channelLeaveHandler = handleLeavePrivateChannel;
     }
-}
+
+    const getIcon = (
```

Comment:
Here you are no longer "getting the icon". This is indeed the icon. So this variable can be renamed as `icon` (or `channelIcon` if you prefer it).